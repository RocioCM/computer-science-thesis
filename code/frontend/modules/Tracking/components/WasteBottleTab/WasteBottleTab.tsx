import React, { useState } from 'react';
import { WasteBottle } from '../../types';
import AccountDetails from '../AccountDetails';
import FaIcon from '@/common/components/FaIcon';

interface Props {
  bottle?: WasteBottle | null;
  options: { label: string; value: number }[];
  handleOption: (optionId: number) => void;
}

const WasteBottleTab: React.FC<Props> = ({ bottle, options, handleOption }) => {
  const [showOptions, setShowOptions] = useState(false);

  return bottle && !showOptions ? (
    <div className="h-full w-full overflow-auto hide-scroll relative">
      {/* Información General */}
      <div className="">
        <div className="space-y-2">
          <div className="w-full flex justify-between items-center">
            <h3 className="text-lg font-semibold border-b pb-2">
              Información General
            </h3>
            {options.length > 1 && (
              <button onClick={() => setShowOptions(!showOptions)}>
                <FaIcon type="fa-solid fa-list" />
              </button>
            )}
          </div>
          <p>
            <span className="font-semibold">Código de seguimiento:</span>{' '}
            {bottle.trackingCode}
          </p>
          <p>
            <span className="font-semibold">Usuario reciclador:</span>{' '}
            <AccountDetails blockchainId={bottle.creator} />
          </p>
          <p>
            <span className="font-semibold">Fecha de reciclaje:</span>{' '}
            {new Date(bottle.createdAt).toLocaleDateString()}
          </p>
        </div>
      </div>
    </div>
  ) : options ? (
    <div className="h-full w-full overflow-auto hide-scroll relative pt-2">
      <h3 className="text-lg font-semibold border-b pb-2">
        Envases reciclados:
      </h3>
      <ul className="mt-2 space-y-2 list-disc list-inside">
        {options.map((option) => (
          <li
            key={option.value}
            onClick={() => {
              setShowOptions(false);
              handleOption(option.value);
            }}
            className="
					cursor-pointer underline"
          >
            <span className="font-semibold">{option.label}</span>
          </li>
        ))}
      </ul>
    </div>
  ) : null;
};

export default WasteBottleTab;
