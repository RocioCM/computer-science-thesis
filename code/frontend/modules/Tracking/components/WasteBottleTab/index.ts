import WasteBottleTab from './WasteBottleTab';

export default WasteBottleTab;
