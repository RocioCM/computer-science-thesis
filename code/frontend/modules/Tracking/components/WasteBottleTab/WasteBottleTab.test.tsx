import { render } from '@testing-library/react';
import WasteBottleTab from './WasteBottleTab';

describe('Tracking - WasteBottleTab', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches snapshot with batch', () => {
    const { container } = render(
      <WasteBottleTab
        bottle={{
          id: 1,
          owner: '0x1234567890abcdef1234567890abcdef12345678',
          creator: '0xabcdef1234567890abcdef1234567890abcdef12',
          recycledBatchId: 1,
          trackingCode: 'TRACK123',
          createdAt: new Date('2025-04-13T13:00:00').toISOString(),
        }}
        options={[]}
        handleOption={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('Matches snapshot with options', () => {
    const { container } = render(
      <WasteBottleTab
        bottle={null}
        options={[
          {
            label: 'Option 1',
            value: 1,
          },
          {
            label: 'Option 2',
            value: 2,
          },
        ]}
        handleOption={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
