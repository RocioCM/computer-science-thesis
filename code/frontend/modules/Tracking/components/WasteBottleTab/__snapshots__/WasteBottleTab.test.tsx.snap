// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking - WasteBottleTab Matches snapshot with batch 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative"
  >
    <div
      class=""
    >
      <div
        class="space-y-2"
      >
        <div
          class="w-full flex justify-between items-center"
        >
          <h3
            class="text-lg font-semibold border-b pb-2"
          >
            Información General
          </h3>
        </div>
        <p>
          <span
            class="font-semibold"
          >
            Código de seguimiento:
          </span>
           
          TRACK123
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Usuario reciclador:
          </span>
           
          <span
            class=""
          >
            0xabcdef1234567890abcdef1234567890abcdef12
          </span>
        </p>
        <p>
          <span
            class="font-semibold"
          >
            Fecha de reciclaje:
          </span>
           
          30/4/2025
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tracking - WasteBottleTab Matches snapshot with options 1`] = `
<div>
  <div
    class="h-full w-full overflow-auto hide-scroll relative pt-2"
  >
    <h3
      class="text-lg font-semibold border-b pb-2"
    >
      Envases reciclados:
    </h3>
    <ul
      class="mt-2 space-y-2 list-disc list-inside"
    >
      <li
        class="      cursor-pointer underline"
      >
        <span
          class="font-semibold"
        >
          Option 1
        </span>
      </li>
      <li
        class="      cursor-pointer underline"
      >
        <span
          class="font-semibold"
        >
          Option 2
        </span>
      </li>
    </ul>
  </div>
</div>
`;
